package/genhackage

```diff
@@ -0,0 +1,231 @@
+    "pandoc": "pandoc",
+    "pandoc-citeproc": "pandoc-citeproc",
+    "pandoc-crossref": "pandoc-crossref",
+    "postgrest": "postgrest",
+    "shellcheck": "ShellCheck",
+    "stack": "stack",
+    "stylish-haskell": "stylish-haskell",
+    "tamarin-prover": "tamarin-prover",
+    "tidalcycles": "tidal",
+    "xmobar": "xmobar",
+    "xmonad": "xmonad",
+    "xmonad-contrib": "xmonad-contrib",
+    "xmonad-utils": "xmonad-utils",
+}
+
+VERSION_SUBSTITUTION = {
+}
+
+URL_PATTERN = "https://www.archlinux.org/packages/%(repo)s/x86_64/%(pkgname)s"
+
+
+def gen_hackage_mapping(repo, quiet=False):
+    counter = 0
+
+    version_mapping = {}
+    arch_package_mapping = {}
+
+    for item in expac("-S", "%P", repo + "/ghc", repo + "/ghc-libs").split():
+        # Fetch package and version
+        package, version = item.split("=")
+
+        # Remove pkgrel
+        version = version.split("-")[0]
+
+        if package in PRESET_PACKAGES:
+            package = PRESET_PACKAGES[package]
+        elif package.startswith("haskell-"):
+            package = package[8:]
+        else:
+            package = False
+
+        if package:
+            if not quiet:
+                print("Found", package, version)
+
+            version_mapping[package] = version
+            arch_package_mapping[package] = "ghc"
+
+            counter += 1
+
+    for line in pacman("-Sl", repo, "--color=never"):
+        # Fetch package and version
+        package, version = line.split()[1:3]
+        arch_package = package
+
+        # Remove pkgrel
+        version = version.split("-")[0]
+
+        if package in PRESET_PACKAGES:
+            package = PRESET_PACKAGES[package]
+        elif package.startswith("haskell-"):
+            package = package[8:]
+        else:
+            package = False
+
+        if package:
+            if not quiet:
+                print("Found", package, version)
+
+            version_mapping[package] = version
+
+            if package not in arch_package_mapping or arch_package in PRESET_PACKAGES:
+                arch_package_mapping[package] = arch_package
+
+            counter += 1
+
+    if not quiet:
+        print(counter, "packages found.")
+
+    return version_mapping, arch_package_mapping
+
+
+def gen_hackage_csv(repo):
+    version_mapping, arch_package_mapping = gen_hackage_mapping(repo)
+    result_fd = StringIO()
+    result_csv = csv.writer(result_fd, quoting=csv.QUOTE_ALL)
+
+    for package, version in version_mapping.items():
+        result_csv.writerow([
+            package,
+            VERSION_SUBSTITUTION.get(version, version),
+            URL_PATTERN % {"pkgname": arch_package_mapping[package], "repo": repo}
+        ])
+
+    result_fd.seek(0)
+    result_body = result_fd.read().rstrip("\r\n")
+    return result_body
+
+
+def submit_hackage(repo):
+    csv_body = gen_hackage_csv(repo)
+    headers={"Content-Type": "text/csv"}
+    if "HACKAGE_USERNAME" in os.environ and "HACKAGE_PASSWORD" in os.environ:
```

Comment:
should the HACKAGE env variables be documented somewhere in the header of the file?